python/pyarrow/_array.pyx

```diff
@@ -992,6 +993,23 @@ cdef class FixedSizeBinaryValue(ArrayValue):
         return cp.PyBytes_FromStringAndSize(data, length)
 
 
+cdef class StructValue(ArrayValue):
+    def as_py(self):
+        cdef:
+            CStructArray* ap
+            vector[shared_ptr[CField]] child_fields = self.type.type.children()
+        ap = <CStructArray*> self.sp_array.get()
+        child_arrays = ap.fields()
+        boxed_arrays = (box_array(child) for child in child_arrays)
+        child_names = (child.get().name() for child in child_fields)
+        # Return the struct as a dict
+        return {
+            name: value for name, value in (
+                (name, child_array[self.index].as_py())
+                for (name, child_array) in
+                izip(child_names, boxed_arrays)
+            ) if value is not None
```

Comment:
Basically to get consistent results with pandas convert, where null values are not being shown. They are converted to `None` by the nested arrays, so i want to filter them.
I think both putting None instead of nulls or not showing them is ok, but it needs to be consistent. I chose eliminating them to potentially reduce memory overhead and to have them  print nicer.